timber-lint/src/main/java/timber/lint/WrongTimberUsageDetector.kt

```diff
@@ -0,0 +1,808 @@
+          Incident(
+            issue = ISSUE_ARG_TYPES,
+            scope = call,
+            location = context.getLocation(argument),
+            message = "Wrong argument type for formatting argument '#${i + 1}' in `${formatString}`: conversion is '`${formatType}`', received `${type.simpleName}` (argument #${startIndexOfArguments + i + 1} in method call)"
+          )
+        )
+      }
+    }
+  }
+
+  private fun getType(expression: UExpression?): Class<*>? {
+    if (expression == null) {
+      return null
+    }
+    if (expression is PsiMethodCallExpression) {
+      val call = expression as PsiMethodCallExpression
+      val method = call.resolveMethod() ?: return null
+      val methodName = method.name
+      if (methodName == GET_STRING_METHOD) {
+        return String::class.java
+      }
+    } else if (expression is PsiLiteralExpression) {
+      val literalExpression = expression as PsiLiteralExpression
+      val expressionType = literalExpression.type
+      when {
+        isString(expressionType!!) -> return String::class.java
+        expressionType === PsiType.INT -> return Integer.TYPE
+        expressionType === PsiType.FLOAT -> return java.lang.Float.TYPE
+        expressionType === PsiType.CHAR -> return Character.TYPE
+        expressionType === PsiType.BOOLEAN -> return java.lang.Boolean.TYPE
+        expressionType === PsiType.NULL -> return Any::class.java
+      }
+    }
+
+    val type = expression.getExpressionType()
+    if (type != null) {
+      val typeClass = getTypeClass(type)
+      return typeClass ?: Any::class.java
+    }
+
+    return null
+  }
+
+  private fun getTypeClass(type: PsiType?): Class<*>? {
+    return when (type?.canonicalText) {
+      null -> null
+      TYPE_STRING, "String" -> String::class.java
+      TYPE_INT -> Integer.TYPE
+      TYPE_BOOLEAN -> java.lang.Boolean.TYPE
+      TYPE_NULL -> Object::class.java
+      TYPE_LONG -> Long.TYPE
+      TYPE_FLOAT -> Float.TYPE
+      TYPE_DOUBLE -> Double.TYPE
+      TYPE_CHAR -> Character.TYPE
+      TYPE_OBJECT -> null
+      TYPE_INTEGER_WRAPPER, TYPE_SHORT_WRAPPER, TYPE_BYTE_WRAPPER, TYPE_LONG_WRAPPER -> Integer.TYPE
+      TYPE_FLOAT_WRAPPER, TYPE_DOUBLE_WRAPPER -> Float.TYPE
+      TYPE_BOOLEAN_WRAPPER -> java.lang.Boolean.TYPE
+      TYPE_BYTE -> Byte.TYPE
+      TYPE_SHORT -> Short.TYPE
+      "Date", "java.util.Date" -> Date::class.java
+      "Calendar", "java.util.Calendar" -> Calendar::class.java
+      "BigDecimal", "java.math.BigDecimal" -> Float.TYPE
+      "BigInteger", "java.math.BigInteger" -> Integer.TYPE
+      else -> null
+    }
+  }
+
+  private fun isSubclassOf(
+    context: JavaContext, expression: UExpression, cls: Class<*>
+  ): Boolean {
+    val expressionType = expression.getExpressionType()
+    if (expressionType is PsiClassType) {
+      return context.evaluator.extendsClass(expressionType.resolve(), cls.name, false)
+    }
+    return false
+  }
+
+  @OptIn(ExperimentalStdlibApi::class)
```

Comment:
What is this used for? It could cause compatibility problems with newer AGPs using newer Kotlin versions. Or even older AGPs and older Kotlin versions. Unless it's absolutely essential we should stick to stable API.

Reply:
For `buildList` which was introduced in 1.3.  I figured this was cleaner, but can revert to a MutableList-for loop implementation.

Reply:
You can replace it with `MutableList().apply { .. }` to get basically the same API on stable Kotlin